### PR TITLE
Issue 209

### PR DESCRIPTION
**Pull request by Tam-Lin**
Think I finally got everything properly signed off on.  Addresses issues #209 and #212.

### CODE REVIEW

**Comment by Lawouach** [comment]
Thank you for the superb analysis and coverage of trying to understand the intent of the code.

Initially, I had never planned on letting nested structures as secrets/configuration. But because I forgot to actively validate against these, they went through the rabbit hole and now it would foolish to not support them. But this may explain why some of these edge case bugs are found.

I think all issues that were rasied are real issues and I'm happy you've made the effort to work on them.

I know our review is late but would it be possible to rebase your branch before we can merge?

Also, my only concern, before I do the thorough review is backward compatibility. I would not agree to a change that breaks existing, specific behavior. Tests seem to pass so that sounds good but I'll review more deeply in a minute.

File: tests/test_secret.py [hunk 1]
```diff
@@ -277,9 +298,133 @@ def test_should_override_load_inline_with_var():
...
+    }, config)
+    assert secrets["myapp"]["token"] == "bar"
+
+@patch('chaoslib.secret.hvac')
+def test_key_not_in_vault(hvac):
+    pass
+
+@patch('chaoslib.secret.hvac')
+def test_no_vault_entry(hvac):
+    pass
```
Comment (ciaranevans) [inline tests/test_secret.py:391]: Hey @Tam-Lin thanks for the PR! @Lawouach is better suited to checking out the majority of the changes, but could you double check to see if you meant to implement these tests?

If so, could they please be implemented, otherwise, could these test definitions be removed as they current don't exercise any functionality 😄 Thanks!

**Review by ciaranevans — CHANGES_REQUESTED** [review]

**Review by Lawouach — CHANGES_REQUESTED** [review]
I think the changes make sense to me and sound clean and keeping compatibility.

We're likely missing a changelog entry and I agree with @ciaranevans, if some tests remain empty, please remove them or implement them :)

**Comment by Tam-Lin** [comment]
Closing this pull request, and will submit new pull request based on latest version.
